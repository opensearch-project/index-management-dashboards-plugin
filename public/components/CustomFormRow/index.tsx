/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */
import React from "react";
import { EuiDescribedFormGroup, EuiFormRow, EuiFormRowProps } from "@elastic/eui";

export type CustomFormRowProps = {
  position?: "top" | "bottom";
  direction?: "hoz" | "ver";
} & EuiFormRowProps;

export default function CustomFormRow(props: CustomFormRowProps) {
  const { helpText, children, position = "top", direction = "ver", label, ...others } = props;
  if (direction === "hoz") {
    return (
      <EuiDescribedFormGroup
        fullWidth
        descriptionFlexItemProps={{ style: { maxWidth: 700 } }}
        title={<div>{label}</div>}
        description={helpText}
      >
        <EuiFormRow {...others}>{children}</EuiFormRow>
      </EuiDescribedFormGroup>
    );
  }

  return (
    <EuiFormRow {...others} label={label} helpText={position === "bottom" ? helpText : undefined}>
      <>
        {helpText && position === "top" ? (
          <div style={{ paddingTop: 0, paddingBottom: 4 }} className="euiFormHelpText euiFormRow__text">
            {helpText}
          </div>
        ) : null}
        {children}
      </>
    </EuiFormRow>
  );
}
