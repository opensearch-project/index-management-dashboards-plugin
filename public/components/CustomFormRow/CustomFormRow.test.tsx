/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render } from "@testing-library/react";
import CustomFormRow from "./index";

describe("<FormGenerator /> spec", () => {
  it("render the component", () => {
    render(
      <CustomFormRow helpText="test">
        <h1>test custom form row</h1>
      </CustomFormRow>
    );
    expect(document.body.children).toMatchSnapshot();
  });
});
