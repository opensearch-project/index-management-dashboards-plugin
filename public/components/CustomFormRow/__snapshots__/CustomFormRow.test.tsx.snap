// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormGenerator /> spec render the component 1`] = `
HTMLCollection [
  <div>
    <div
      class="euiFormRow"
      id="some_html_id-row"
    >
      <div
        class="euiFormRow__fieldWrapper"
      >
        <div
          class="euiFormHelpText euiFormRow__text"
          style="padding-top: 0px; padding-bottom: 4px;"
        >
          test
        </div>
        <h1>
          test custom form row
        </h1>
      </div>
    </div>
  </div>,
]
`;
